policy/handler.go

```diff
@@ -0,0 +1,258 @@
+package policy
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/atomist-skills/go-skill"
+	"github.com/atomist-skills/go-skill/policy/evaluators"
+	"github.com/atomist-skills/go-skill/policy/evaluators/data"
+	"github.com/atomist-skills/go-skill/policy/goals"
+	"github.com/atomist-skills/go-skill/policy/query"
+	"github.com/atomist-skills/go-skill/policy/storage"
+	"github.com/atomist-skills/go-skill/util"
+	"olympos.io/encoding/edn"
+)
+
+type EvaluatorSelector func(ctx context.Context, req skill.RequestContext, goal *goals.Goal, dataSource data.DataSource) (evaluators.GoalEvaluator, error)
+
+var evaluatorSelector EvaluatorSelector
+
+func StartPolicy(
+	subscriptionNames []string,
+	selector EvaluatorSelector,
+) {
+	evaluatorSelector = selector
+	handlers := skill.Handlers{
+		"async-query-packages":      evaluateGoalsWithAsyncQueryResult,
+		"async-query-image-details": evaluateGoalsWithAsyncQueryResult,
```

Comment:
not-blocking: The non-root-user skill doesn't need these 2 handlers. Can we have the caller pass the ones that are needed to the `StartPolicy` func?